packages/video_player_videohole/tizen/src/video_player.cc

```diff
@@ -0,0 +1,547 @@
+
+  return player_id_;
+}
+
+void VideoPlayer::SetDisplayRoi(int32_t x, int32_t y, int32_t width,
+                                int32_t height) {
+  int ret = player_set_display_roi_area(player_, x, y, width, height);
+  if (ret != PLAYER_ERROR_NONE) {
+    LOG_ERROR("[VideoPlayer] player_set_display_roi_area failed: %s",
+              get_error_message(ret));
+  }
+}
+
+VideoPlayer::~VideoPlayer() {
+  if (drm_manager_) {
+    drm_manager_->ReleaseDrmSession();
+  }
+  Dispose();
+}
+
+void VideoPlayer::Play() {
+  LOG_INFO("[VideoPlayer] Player starting.");
+
+  player_state_e state = PLAYER_STATE_NONE;
+  int ret = player_get_state(player_, &state);
+  if (ret != PLAYER_ERROR_NONE) {
+    LOG_ERROR("[VideoPlayer] Unable to get player state.");
+    return;
+  }
+  if (state == PLAYER_STATE_NONE || state == PLAYER_STATE_IDLE) {
+    LOG_ERROR("[VideoPlayer] Player not ready.");
+    return;
+  }
+  if (state == PLAYER_STATE_PLAYING) {
+    LOG_INFO("[VideoPlayer] Player already playing.");
+    return;
+  }
+
+  ret = player_start(player_);
+  if (ret != PLAYER_ERROR_NONE) {
+    LOG_ERROR("[VideoPlayer] player_start failed: %s", get_error_message(ret));
+  }
+}
+
+void VideoPlayer::Pause() {
+  LOG_INFO("[VideoPlayer] Player pausing.");
+
+  player_state_e state = PLAYER_STATE_NONE;
+  int ret = player_get_state(player_, &state);
+  if (ret != PLAYER_ERROR_NONE) {
+    LOG_ERROR("[VideoPlayer] Unable to get player state.");
+    return;
+  }
+  if (state == PLAYER_STATE_NONE || state == PLAYER_STATE_IDLE) {
+    LOG_ERROR("[VideoPlayer] Player not ready.");
+    return;
+  }
+  if (state != PLAYER_STATE_PLAYING) {
+    LOG_INFO("[VideoPlayer] Player not playing.");
+    return;
+  }
+
+  ret = player_pause(player_);
+  if (ret != PLAYER_ERROR_NONE) {
+    LOG_ERROR("[VideoPlayer] player_pause failed: %s", get_error_message(ret));
+  }
+}
+
+void VideoPlayer::SetLooping(bool is_looping) {
+  LOG_INFO("[VideoPlayer] is_looping: %d", is_looping);
+
+  int ret = player_set_looping(player_, is_looping);
+  if (ret != PLAYER_ERROR_NONE) {
+    LOG_ERROR("[VideoPlayer] player_set_looping failed: %s",
+              get_error_message(ret));
+  }
+}
+
+void VideoPlayer::SetVolume(double volume) {
+  LOG_INFO("[VideoPlayer] volume: %f", volume);
+
+  int ret = player_set_volume(player_, volume, volume);
+  if (ret != PLAYER_ERROR_NONE) {
+    LOG_ERROR("[VideoPlayer] player_set_volume failed: %s",
+              get_error_message(ret));
+  }
+}
+
+void VideoPlayer::SetPlaybackSpeed(double speed) {
+  LOG_INFO("[VideoPlayer] speed: %f", speed);
+
+  int ret = player_set_playback_rate(player_, speed);
+  if (ret != PLAYER_ERROR_NONE) {
+    LOG_ERROR("[VideoPlayer] player_set_playback_rate failed: %s",
+              get_error_message(ret));
+  }
+}
+
+void VideoPlayer::SeekTo(int32_t position) {
+  LOG_INFO("[VideoPlayer] position: %d", position);
+
+  int ret = player_set_play_position(player_, position, true, nullptr, nullptr);
```

Comment:
Why you remove OnSeekCompleted()?
I know I didn't set the reply callback in SeekTo(), but we can add it and assign `on_completed_cb_` before calling 'player_set_play_position'.
Maybe add the callback is better than remove it?

Reply:
Removed because it was unused. Would you like to make `SeekTo` an async call just like [video_player_tizen](https://github.com/flutter-tizen/plugins/blob/510789f0c5b9594b02604b349450a22c02669142/packages/video_player/pigeons/messages.dart#L65-L66)?

Reply:
> Would you like to make `SeekTo` an async call just like [video_player_tizen](https://github.com/flutter-tizen/plugins/blob/510789f0c5b9594b02604b349450a22c02669142/packages/video_player/pigeons/messages.dart#L65-L66)?

Yes, I did not notice it before. Maybe It's better to add the callback.